src/OrchardCore.Modules/OrchardCore.Users/TwoFactorAuthenticationStartup.cs

```diff
@@ -96,10 +94,13 @@ public class EmailAuthenticatorStartup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.Configure<TwoFactorOptions>(options =>
-        {
-            options.Providers.Add(TokenOptions.DefaultEmailProvider);
-        });
+        var emailProviderType = typeof(EmailTokenProvider<>).MakeGenericType(typeof(IUser));
+        services.AddTransient(emailProviderType)
```

Comment:
@sebastienros the reason we create the type using `MakeGenericType` here is because we also need the type on line 102 when creating an instance of `TokenProviderDescriptor`.

Reply:
so again `typeof(EmailTokenProvider<IUser>)` on line 102?

Reply:
yes but why? Check this out https://github.com/dotnet/aspnetcore/blob/main/src/Identity/Core/src/IdentityBuilderExtensions.cs#L28-L38

Reply:
@sebastienros do you still need me to update this?